# import statements

In [43]:
import pandas as pd
import numpy as np

# code to read student.csv into pandas dataframe 

In [46]:
student = pd.read_csv('student.csv', delimiter = ';')
student.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

# helper functions (if needed)

In [16]:
#python code...

# Python code to answer question 1) What is the minimum, maximum, average, and median final grade (G3 column)?

In [51]:
columnG3 = student['G3']
minG3 = np.min(columnG3)
print("min: ", minG3)
maxG3 = np.max(columnG3)
print("max: ", maxG3)
avgG3 = np.mean(columnG3)
print("average: ", avgG3)
medianG3 = np.median(columnG3)
print("median: ", medianG3)

min:  0
max:  20
average:  10.415189873417722
median:  11.0


# Text answer for question 1.

minimum: 0
maximum: 20
average: 10.415189873417722
median: 11.0

# Python code to answer question 2) Which subgroup of sex, male or female, had the highest average final grade (G3)?

In [62]:
female = (student['sex'] == 'F')
final_f = student.loc[female, 'G3']
avgFFG = np.mean(final_f)
print("female avg: ", avgFFG)

male = (student['sex'] == 'M')
final_m = student.loc[male, 'G3']
avgMFG = np.mean(final_m)
print("male avg: ", avgMFG)

female avg:  9.966346153846153
male avg:  10.914438502673796


# Text answer for question 2.  

Male

# Python code to answer question 3) Which subgroup of Fjob had the lowest average final grade?

In [69]:
at_home = (student['Fjob'] == 'at_home')
final_atHome = student.loc[at_home, 'G3']
avgAhFG = np.mean(final_atHome)
print("at home avg: ", avgAhFG)

health = (student['Fjob'] == 'health')
final_health = student.loc[health, 'G3']
avgHFG = np.mean(final_health)
print("health avg: ", avgHFG)

other = (student['Fjob'] == 'other')
final_other = student.loc[other, 'G3']
avgOFG = np.mean(final_other)
print("other avg: ", avgOFG)

services = (student['Fjob'] == 'services')
final_services = student.loc[services, 'G3']
avgSFG = np.mean(final_services)
print("services avg: ", avgSFG)

teacher = (student['Fjob'] == 'teacher')
final_teacher = student.loc[teacher, 'G3']
avgTFG = np.mean(final_teacher)
print("teacher avg: ", avgTFG)

at home avg:  10.15
health avg:  11.61111111111111
other avg:  10.193548387096774
services avg:  10.297297297297296
teacher avg:  11.96551724137931


# Text answer for question 3.

at_home Fjob lowest average final grade

# Python code to answer question 4) On average, did students in romantic relationships get higher or lower final grades?

In [71]:
relationship = (student['romantic'] == 'yes')
relationStatus = student.loc[relationship, 'G3']
avgRrFG = np.mean(relationStatus)
print("romanic relationship avg: ", avgRrFG)

alone = (student['romantic'] == 'no')
aloneAffect = student.loc[alone, 'G3']
avgAFG = np.mean(aloneAffect)
print("no romantic relationship avg: ", avgAFG)

romanic relationship avg:  9.575757575757576
no romantic relationship avg:  10.836501901140684


# Text answer for question 4.

lower

# Python code to answer question 5) On average, do people with internet access at home make higher or lower final grades than people without internet? 

In [73]:
internet = (student['internet'] == 'yes')
internetStatus = student.loc[internet, 'G3']
avgIFG = np.mean(internetStatus)
print("internet access avg: ", avgIFG)

no_internet = (student['internet'] == 'no')
internetState = student.loc[no_internet, 'G3']
avgNiFG = np.mean(internetState)
print("No internet access avg: ", avgNiFG)

internet access avg:  10.617021276595745
No internet access avg:  9.409090909090908


# Text answer for question 5.

higher

# Python code to answer question 6) What percentage of male students are in romantic relationships?  What percentage of female students are in romantic relationships?

In [22]:
#python code...

# Text answer for question 6.

text...

# Python code to answer question 7) What is the correlation between freetime and final grade? 

In [82]:
columnFt = student['freetime']
cor_ftFG = np.corrcoef(columnFt, columnG3)
print(cor_ftFG)

[[1.         0.01130724]
 [0.01130724 1.        ]]


# Text answer for question 7.

0.01130724

# Python code to answer question 8) What is the correlation between studytime and final grade? 

In [83]:
columnSt = student['studytime']
cor_stFG = np.corrcoef(columnSt, columnG3)
print(cor_stFG)

[[1.         0.09781969]
 [0.09781969 1.        ]]


# Text answer for question 8.

0.09781969

# Python code to answer question 9) What is the correlation between studytime and freetime? 

In [84]:
cor_stFT = np.corrcoef(columnSt, columnFt)
print(cor_stFT)

[[ 1.         -0.14319841]
 [-0.14319841  1.        ]]


# Text answer for question 9.

-0.14319841

# Python code to answer question 10) Excluding columns G1 and G2, which numerical column had the biggest correlation with final grade (G3)?

In [26]:
#python code...

# Text answer for question 10.  Please identify the column and correlation.

text...